openlibrary/plugins/wikidata/code.py

```diff
@@ -0,0 +1,45 @@
+"""
+The purpose of this file is to:
+1. Interact with the Wikidata API
+2. Store the results
+3. Make the results easy to access from other files
+"""
+import requests
+from dataclasses import dataclass
+from openlibrary.core.helpers import seconds_since
+
+from openlibrary.core.wikidata import WikidataEntities
+
+MONTH_IN_SECONDS = 60 * 60 * 24 * 30
+
+
+@dataclass
+class WikiDataEntity:
+    id: str
+    descriptions: dict[str, str]
+
+    def description(self, language: str = 'en') -> str | None:
+        """If a description isn't available in the requested language default to English"""
+        return self.descriptions.get(language) or self.descriptions.get('en')
+
+
+def get_wikidata_entity(QID: str, ttl: int = MONTH_IN_SECONDS) -> WikiDataEntity | None:
+    """
+    This only supports QIDs, if we want to support PIDs we need to use different endpoints
+    ttl (time to live) inspired by the cachetools api https://cachetools.readthedocs.io/en/latest/#cachetools.TTLCache
+    """
+    entity = WikidataEntities.get_by_id(QID)
+    if entity and seconds_since(entity.updated) < ttl:
```

Comment:
We can probably inline this function, and the `timedelta` helpers might be a little clearer here!

```suggestion
    if entity and (datetime.now() - entity.updated) < timedelta(days=30):
```